From 5d60c926219d683921b022738ac54aafd9b0d060 Mon Sep 17 00:00:00 2001
From: Liu Ping Fan <kernelfans@gmail.com>
Date: Thu, 21 Nov 2013 10:17:54 +0800
Subject: [PATCH] powerpc/mm: use macro PGTABLE_EADDR_SIZE instead of digital

References: fate#322495, bsc#1032471
Patch-mainline: v3.16-rc1
Git-commit: 5a4e58bc693f04aa650219784e5e339e0db6b902

In case of extending the eaddr in future, use this macro
PGTABLE_EADDR_SIZE to ease the maintenance of the code.

Signed-off-by: Liu Ping Fan <pingfank@linux.vnet.ibm.com>
Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Acked-by: Michal Suchanek <msuchanek@suse.de>
---
 arch/powerpc/mm/slb_low.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/powerpc/mm/slb_low.S b/arch/powerpc/mm/slb_low.S
index 996770577d35..7209ce572f6f 100644
--- a/arch/powerpc/mm/slb_low.S
+++ b/arch/powerpc/mm/slb_low.S
@@ -77,7 +77,7 @@ _GLOBAL(slb_allocate_realmode)
 	 * check for bad kernel/user address
 	 * (ea & ~REGION_MASK) >= PGTABLE_RANGE
 	 */
-	rldicr. r9,r3,4,(63 - 46 - 4)
+	rldicr. r9,r3,4,(63 - PGTABLE_EADDR_SIZE - 4)
 	bne-	8f
 
 	srdi	r9,r3,60		/* get region */
-- 
2.10.2

